### ISSUE TITLE
Sentinel-1 problem of reshaping geolocation grids

### ISSUE THREAD

**Issue by knutfrode** [nansencenter/nansat#195]
For some S1-files, there is a problem in reshaping lon,lat,inc,ele arrays to shape (len(Y), len(X)), as size of geolocation grids. This leads to error: ``` Traceback (most recent call last): File "/home/knutfd/software/nansat/utilities/nansatinfo", line 18, in n = Nansat(sys.argv[1]) File "/home/knutfd/software/nansat/nansat/nansat.py", line 166, in __init__ self.vrt = self._get_mapper(mapperName, **kwargs) File "/home/knutfd/software/nansat/nansat/nansat.py", line 1715, in _get_mapper **kwargs) File "/home/knutfd/software/nansat/nansat/mappers/mapper_sentinel1_l1.py", line 178, in __init__ lon = np.array(lon).reshape(sh) ValueError: cannot reshape array of size 399 into shape (19,41) ``` This can be reproduced with this file: https://drive.google.com/open?id=0B2jJyde278m3ZU9raEdNSEpid00 The following fix in Sentinel1-mapper solves the problem for some scenes, by reducing either dimension by one, but in some cases there is a larger mismatch, which is not clear to me. ``` - - lon = np.array(lon).reshape(len(Y), len(X)) - lat = np.array(lat).reshape(len(Y), len(X)) - inc = np.array(inc).reshape(len(Y), len(X)) - ele = np.array(ele).reshape(len(Y), len(X)) + nX = len(X) + nY = len(Y) + nE = lon.shape[0] + sh = (nY, nX) + if nE != nX*nY: + if (nX-1)*nY == nE: + sh = (nY, nX-1) + elif (nX)*(nY-1) == nE: + sh = (nY-1, nX) + elif (nX+1)*nY == nE: + sh = (nY, nX+1) + elif nX*(nY+1) == nE: + sh = (nY+1, nX) + if sh[0]*sh[1] == nE: + print 'Changed shape!' + + lon = np.array(lon).reshape(sh) + lat = np.array(lat).reshape(sh) + inc = np.array(inc).reshape(sh) + ele = np.array(ele).reshape(sh) ```

**Comment by jeong1park** [comment]
This is because the geolocation grids are not always sampled in a regular interval. 
Since the width and length of the geolocation grids are calculated by taking the number of unique values from X and Y as in lines 159-160 of "mapper_sentinel1_l1.py", sometimes the resulting values does not match the true dimensions in the original metadata. 
This bug can be fixed simply by calculating the number of "0" in X and Y. 
In short, the reshaping dimensions (len(Y), len(X)) in lines 162-165 should be replaced by ((X==0).sum(), (Y==0).sum()), and lines 159-160 should be deleted.
